src/Repository/EventRepository.php

```diff
@@ -113,4 +113,149 @@ public function countItemsAdmin()
             ->getQuery()
             ->getSingleScalarResult();
     }
+
+    /**
+     * @param $limit
+     * @param $offset
+     */
+    public function findListAdminToApprove($limit, $offset)
+    {
+        return $this->createQueryBuilder('event')
+            ->andWhere('event.status = :val')
+            ->setParameter('val', Event::STATUS_PENDING)
+            ->orderBy('event.startDate', 'ASC')
+            ->setFirstResult($offset)
+            ->setMaxResults($limit)
+            ->getQuery()
+            ->getResult();
+    }
+
+    /**
+     * @return integer
+     */
+    public function countItemsAdminToApprove()
+    {
+        return $this->createQueryBuilder('event')
+            ->select('count(event.id)')
+            ->andWhere('event.status = :val')
+            ->setParameter('val', Event::STATUS_PENDING)
+            ->getQuery()
+            ->getSingleScalarResult();
+    }
+
+    /**
+     * @param $limit
+     * @param $offset
+     * @param $params
+     */
+    public function findListByCriterias($params)
+    {
+        $qb = $this->setCriterias($params);
+
+        return $qb
+            ->orderBy('event.startDate', 'ASC')
+            ->getQuery()
+            ->getResult();
+    }
+
+    /**
+     * @return integer
+     */
+    public function countItemsByCriterias($params)
+    {
+        $qb = $this->setCriterias($params);
+
+        return $qb
+            ->select('count(event.id)')
+            ->getQuery()
+            ->getSingleScalarResult();
+    }
+
+    public function setCriterias($params)
+    {
+        $qb = $this->createQueryBuilder('event')
+            ->innerJoin('event.subgenre', 'subgenre')
```

Comment:
@daria-sieroshtan 
please check that this fields always exists

Reply:
always exists, is validated on assert